static/panes/compiler.js

```diff
@@ -130,33 +129,77 @@ function Compiler(hub, container, state) {
 
     this.fontScale = new FontScale(this.domRoot, state, this.outputEditor);
 
-    this.compilerPicker.selectize({
+    var self = this;
+    var optgroups = this.compilerService.getGroupsInUse(this.currentLangId);
+    optgroups.unshift({value:'favs',label:'Favorites'});
+
+    this.compilerSelectizer = new TomSelect(this.compilerPicker[0],{
         sortField: this.compilerService.getSelectizerOrder(),
         valueField: 'id',
         labelField: 'name',
         searchField: ['name'],
         optgroupField: 'group',
-        optgroups: this.compilerService.getGroupsInUse(this.currentLangId),
+        optgroups: optgroups,
         lockOptgroupOrder: true,
         options: _.filter(this.getCurrentLangCompilers(), function (e) {
             return !e.hidden || e.id === state.compiler;
         }),
         items: this.compiler ? [this.compiler.id] : [],
         dropdownParent: 'body',
         closeAfterSelect: true,
-    }).on('change', _.bind(function (e) {
-        var val = $(e.target).val();
-        if (val) {
-            ga.proxy('send', {
-                hitType: 'event',
-                eventCategory: 'SelectCompiler',
-                eventAction: val,
-            });
-            this.onCompilerChange(val);
+        plugins:['dropdown_input'],
+        onChange: function (val){
+            if (val) {
+                ga.proxy('send', {
+                    hitType: 'event',
+                    eventCategory: 'SelectCompiler',
+                    eventAction: val,
+                });
+                self.onCompilerChange(val);
+            }
+        },
+        duplicates: true,
+        render:{
+            option:function (data,escape){
+                return '<div class="d-flex"><div>'
+					+ escape(data.name)
+					+ '</div><div class="ml-auto toggle-fav"><i class="fas fa-star"></i></div></div>';
+            },
+        },
+    });
+
+    $(this.compilerSelectizer.dropdown_content).on('click','.toggle-fav',function (evt){
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        var option_el        = this.closest('.option');
+        var value            = option_el.dataset.value;
+        var data            = self.compilerSelectizer.options[value];
+        var remove_fav        = false;
+
+        data.group = Array.isArray(data.group) ? data.group : [data.group];
+
+        if( option_el.classList.contains('fav') ){
```

Comment:
In general if you could try and stick to the (unwritten!) style, that would be great. We're missing automatic formatting, which I should Just Fix rather than have to tell you about it, sorry. Ideally spaces before `{` in like `function (evt) {` and space after if and not after paren (`if (option_el...`). But it's low priority, currently the JB IDE settings are the way we fix this. I'm happy to do so after...